contexts/MembershipContext/src/Domain/Model/Application.php

```diff
@@ -166,4 +166,16 @@ public function isDeleted(): bool {
 		return $this->isDeleted;
 	}
 
+	public function setFirstPaymentDate( string $firstPaymentDate ) {
+		/** @var PayPalPayment $payPalPayment */
+		$payPalPayment = $this->getPayment()->getPaymentMethod();
+		$payPalData = $payPalPayment->getPayPalData();
+		if ( $payPalData === null ) {
+			$payPalData = new PayPalData();
+		}
+
+		$payPalData->setFirstPaymentDate( $firstPaymentDate );
+		$this->addPayPalData( $payPalData );
```

Comment:
Some questions related to this that probably should not be acted upon in this PR:

Why can PayPalPayment have a null PayPalData? I'm seeing some code that is giving it an empty PayPalData, so can't a PayPalData be required? If not for some reason, could the getPayPalData method not always return a PayPalData anyway (by creating an empty instance if the field is null)?

Why do we have a PayPalData class in the first place? PayPalPayment looks like a rather empty container like this. Could the stuff that is in PayPalData not be in PayPalPayment?